src/shared/components/ResourceEditor/useEditorTooltip.tsx

```diff
@@ -0,0 +1,405 @@
+import * as React from 'react';
```

Comment:
I think the functions inside this file, especially the two hooks, should be unit tested against different scenarios.

Reply:
agreed, will be in different PR if you don't mind